src/OrchardCore/OrchardCore.Mvc.Core/SharedViewCompilerProvider.cs

```diff
@@ -20,6 +21,10 @@ namespace OrchardCore.Mvc
 {
     public class SharedViewCompilerProvider : IViewCompilerProvider
     {
+        // We use a static 'MemoryCache' so that it is shared across tenants. We don't need a custom
+        // 'IViewCompilationMemoryCacheProvider' because we are overriding 'RazorViewCompilerProvider'.
```

Comment:
> Because we override `RazorViewCompilerProvider`

But the point of this new `IViewCompilationMemoryCacheProvider` is that we don't need to override it anymore, and copy paste their code. Or do we still need it anyway?

Reply:
No, it allowed to remove `SharedViewCompiler` the biggest and more complex class we added to share the cache, but we still need `SharedViewCompilerProvider` which was created well before.

In the last one, e.g we adapt the relative path of module precompiled views.

Reply:
Hmm, maybe we created the 2 classes at the same time, but we need the last one.